c7n/resources/iam.py

```diff
@@ -37,14 +37,236 @@ class Role(QueryResourceManager):
 
 @resources.register('iam-user')
 class User(QueryResourceManager):
-
     resource_type = 'aws.iam.user'
 
 
+@resources.register('iam-policy')
+class Policy(QueryResourceManager):
+    resource_type = 'aws.iam.policy'
+
+
+@resources.register('iam-profile')
+class InstanceProfile(QueryResourceManager):
+    resource_type = 'aws.iam.instance-profile'
+
+
+@resources.register('iam-certificate')
+class ServerCertificate(QueryResourceManager):
+    resource_type = 'aws.iam.server-certificate'
+
+
+class IamRoleUsage(Filter):
+
+    def get_service_roles(self):
+        results = []
+        for result in self.scan_lambda_roles():
+            if result not in results:
+                results.append(result)
+        for result in self.scan_ecs_roles():
+            if result not in results:
+                results.append(result)
+        for result in self.scan_asg_roles():
+            if result not in results:
+                results.append(result)
+        for result in self.scan_ec2_roles():
+            if result not in results:
+                results.append(result)
+        return results
+
+    def get_instance_profiles(self):
```

Comment:
this method name could be clearer, get_instance_profiles sounds like it could be the same as 'iam.list_instance_profiles', its good to distinguish that its `used` or extant instance profiles.